webpack/webpack.common.js

```diff
@@ -0,0 +1,50 @@
+const NODE_ENV = process.env.NODE_ENV || "";
+
+require("dotenv").config({ path: `${NODE_ENV}.env` });
+
+const path = require("path");
+const webpack = require("webpack");
+const HtmlWebpackPlugin = require("html-webpack-plugin");
+
+const jsLoaders = require("./loaders/js-loaders");
+const cssLoaders = require("./loaders/css-loaders");
+
+module.exports = {
+  target: "web",
+  stats: {
+    errorDetails: true,
+  },
+  entry: {
+    config: path.join(__dirname, "../src/config.js"),
+    app: path.join(__dirname, "../src/index.js"),
+  },
+  output: {
+    publicPath: "/",
+    filename: "[name].bundle.js",
+    path: path.join(__dirname, "../dist"),
+    chunkFilename: "[name]-[id].js",
+    clean: true,
+  },
+  module: {
+    rules: [
+      {
+        test: /\.(css|scss)$/i,
+        use: cssLoaders,
+      },
+      {
+        test: /\.(js)?$/,
+        use: jsLoaders,
+        exclude: [/(node_modules)/],
+      },
+    ],
+  },
+  plugins: [
+    new webpack.DefinePlugin({
+      "process.env.BACKEND_URL": JSON.stringify(process.env.BACKEND_URL),
```

Comment:
**Issue**: Hardcoding environment variables in the Webpack config can expose sensitive keys and is not recommended for production environments. <br> **Fix**: Use environment variables securely by accessing them at runtime in the application, not at build time. <br> **Code Suggestion**: 
 ```
 -      "process.env.STRIPE_API_KEY": JSON.stringify(process.env.STRIPE_API_KEY),
 +      // Removed for security reasons. Access this variable at runtime in the application, not at build time.
 ```